packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts

```diff
@@ -786,6 +786,18 @@ export class LaunchTemplate extends Resource implements ILaunchTemplate, iam.IGr
     }
   }
 
+  /**
+   * Add the security group to the instance.
+   *
+   * @param securityGroup: The security group to add
+   */
+  public addSecurityGroup(securityGroup: ISecurityGroup): void {
+    if (!this._connections) {
+      throw new Error('LaunchTemplate can only be added a securityGroup if another securityGroup is initialized in the constructor.');
```

Comment:
Why?

Reply:
Thank you for your comment.
Because I understood that `_connections` is initialized only in the constructor.
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts#L585

If `_connections` wasn't initialized in the constructor, an error occurs when trying to call `addSecurityGroup` here.